### PR TITLE
Receive AchievementUpdates from RabbitMQ

### PR DESCRIPTION
**Pull request by Brutus5000**
Using RabbitMQ for unidirectional messages has multiple advantages:
- No messages get lost if the API is not available
- Error messages get parked in a dead letter queue, so they can be reprocessed after fixing
- No handling of server-to-server OAuth handling (permissions to RabbitMQ are implicit)

### CODE REVIEW

File: src/main/java/com/faforever/api/config/RabbitConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    factory.setMessageConverter(new Jackson2JsonMessageConverter());
+
+    return factory;
+  }
+
+  /**
+   * Declare all rabbitmq objects
+   */
+  @Bean
+  public Declarables declarables() {
```
Comment (bukajsytlos) [inline src/main/java/com/faforever/api/config/RabbitConfiguration.java:76]: does this need to be public when other `@Bean` are not?

File: src/main/java/com/faforever/api/config/RabbitConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+      exchanges,
+      queues,
+      bindWithDlq(fafLobbyExchange, deadLetterExchange, achievementQueue, QUEUE_ACHIEVEMENT_ROUTING_KEY)
+    );
+  }
+
+  /**
+   * Creates a binding for a queue along with a matching DLQ and DLQ binding
+   */
+  public List<Declarable> bindWithDlq(Exchange fromExchange, Exchange dlqExchange, Queue queue, String routingKey) {
```
Comment (bukajsytlos) [inline src/main/java/com/faforever/api/config/RabbitConfiguration.java]: could this be private?

File: src/main/java/com/faforever/api/config/RabbitConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+      .bind(dlq)
+      .to(dlqExchange)
+      .with(routingKey)
+      .and(Map.of("x-dead-letter-exchange", dlqExchange.getName()));
+
+    return List.of(queueBinding, dlq, dlqBinding);
+  }
+
+  @SafeVarargs
+  public final Declarables merge(List<Declarable>... declarables) {
```
Comment (bukajsytlos) [inline src/main/java/com/faforever/api/config/RabbitConfiguration.java]: this also (private)

**Review by bukajsytlos — DISMISSED** [review]

File: src/main/java/com/faforever/api/config/RabbitConfiguration.java (hunk 3 above)
Comment (Brutus5000) [reply src/main/java/com/faforever/api/config/RabbitConfiguration.java]: correct. also final was stupid :D

File: src/main/java/com/faforever/api/config/RabbitConfiguration.java (hunk 2 above)
Comment (Brutus5000) [reply src/main/java/com/faforever/api/config/RabbitConfiguration.java]: yes

File: src/main/java/com/faforever/api/config/RabbitConfiguration.java (hunk 1 above)
Comment (Brutus5000) [reply src/main/java/com/faforever/api/config/RabbitConfiguration.java:76]: I checked the other configs and we went for all beans are public. so I made the other ones public instead

**Review by bukajsytlos — DISMISSED** [review]
lgtm

**Comment by Askaholic** [comment]
So what are the routing keys for achievements and events? And how do we get the confirmation responses?

**Pull request merged**